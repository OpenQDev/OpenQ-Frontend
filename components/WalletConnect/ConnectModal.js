import React, { useEffect, useContext } from 'react';
import { metaMask, walletConnect, uauth } from './connectors';
import useWeb3 from '../../hooks/useWeb3';
import Image from 'next/image';
import ModalLarge from '../Utils/ModalLarge';
import StoreContext from '../../store/Store/StoreContext';

const ConnectModal = ({ closeModal, setShowModal }) => {
  const { account } = useWeb3();
  const [appState] = useContext(StoreContext);
  const { accountData } = appState;

  const handleMetaMask = async () => {
    try {
      if (!metaMask) return;
      await metaMask.activate();
      closeModal();
    } catch (err) {
      appState.logger.info(err, accountData?.id, 'ConnectModal.js');
    }
  };

  const handleWalletConnect = async () => {
    try {
      if (!walletConnect) return;
      await walletConnect.activate();
      closeModal();
    } catch (err) {
      appState.logger.info(err, accountData?.id, 'ConnectModal.js');
    }
  };

  const handleUnstoppableConnect = async () => {
    await uauth.activate();
    closeModal();
  };

  useEffect(() => {
    if (account) {
      closeModal();
    }
  }, [account]);

  const btn = (
    <button onClick={closeModal} className='btn-default w-full'>
      Close
    </button>
  );

  return (
    <ModalLarge
      isWalletConnect={true}
      title={'Connect Wallet'}
      footerRight={btn}
      setShowModal={setShowModal}
      resetState={closeModal}
    >
      <div className='p-4'>
        <p className='text-xl pt-2'>
          Connect your wallet to continue with OpenQ. By connecting your wallet you agree with OpenQ{"'"}s{' '}
          <a className='hover:underline text-blue-400' target='_blank' rel='noopener noreferrer' href='/terms-of-use'>
            terms of service
          </a>
          .
        </p>
        <button onClick={handleMetaMask} className='flex py-4 pl-4 my-8 w-full gap-8 btn-default'>
          <Image src={'/wallet-logos/metamask.png'} height={40} width={40} alt={'metamask logo'} />
          <div className='text-lg leading-loose'>Metamask</div>
        </button>
        <button onClick={handleWalletConnect} className='flex py-4 pl-4 mb-8 w-full gap-8 btn-default'>
          <Image
            src={'/wallet-logos/wallet-connect.jpg'}
            className='rounded-full'
            height={40}
            width={40}
            alt={'wallet connect logo'}
          />
          <div className='leading-loose text-lg'>WalletConnect</div>
        </button>
      </div>
    </ModalLarge>
  );
};
export default ConnectModal;
