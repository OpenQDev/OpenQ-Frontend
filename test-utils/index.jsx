// Third party
import React from 'react';
// Cusotm
import { render, renderHook } from '@testing-library/react';
import { cleanup } from '@testing-library/react';

import StoreProvider from '../store/Store/TestStoreProvider';
import AuthProvider from '../store/AuthStore/TestAuthProvider';
import FundProvider from '../components/FundBounty/FundStore/TestFundProvider';
import TokenProvider from '../components/TokenSelection/TokenStore/TokenProvider';

// Add in any providers here if necessary:
// test-utils.js
// (ReduxProvider, ThemeProvider, etc)
afterEach(() => {
  cleanup();
});

const customRender = (ui, options = {}, storeProps, authProps) => {
  const IntersectionObserverMock = vi.fn(() => ({
    disconnect: vi.fn(),
    observe: vi.fn(),
    takeRecords: vi.fn(),
    unobserve: vi.fn(),
  }));
  vi.stubGlobal('IntersectionObserver', IntersectionObserverMock);
  const Providers = ({ children }) => {
    return (
      <AuthProvider StoreProps={authProps}>
        <StoreProvider StoreProps={storeProps}>
          <TokenProvider>{children}</TokenProvider>
        </StoreProvider>
      </AuthProvider>
    );
  };

  return render(ui, { wrapper: Providers, ...options });
};
const customHookRender = (hook, options = {}, storeProps, authProps, fundProps) => {
  const Providers = ({ children }) => {
    return (
      <AuthProvider StoreProps={authProps}>
        <StoreProvider StoreProps={storeProps}>
          <FundProvider StoreProps={fundProps}>
            <TokenProvider>{children}</TokenProvider>
          </FundProvider>
        </StoreProvider>
      </AuthProvider>
    );
  };

  return renderHook(hook, { wrapper: Providers, ...options });
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render, customHookRender as hookRender };
