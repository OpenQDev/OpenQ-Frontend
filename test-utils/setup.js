import { expect, beforeEach } from 'vitest';
import { cleanup } from '@testing-library/react';
import matchers from '@testing-library/jest-dom/matchers';

expect.extend(matchers);
beforeEach(() => {
  cleanup();
});
process.on('unhandledRejection', (reason) => {
  // eslint-disable-next-line no-console
  console.log(`FAILED TO HANDLE PROMISE REJECTION`);
  throw reason;
});
