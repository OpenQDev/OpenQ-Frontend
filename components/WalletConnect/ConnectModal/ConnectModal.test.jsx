// test/component/WatchButton/WatchButton.test
/**
 * @vi-environment jsdom
 */
import React from 'react';
import ConnectModal from '.';
import { render } from '../../../test-utils';

describe('ConnectModal', () => {
  it('should render match DOM Snapshot', () => {
    const { asFragment } = render(<ConnectModal closeModal={() => {}} setShowModal={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
