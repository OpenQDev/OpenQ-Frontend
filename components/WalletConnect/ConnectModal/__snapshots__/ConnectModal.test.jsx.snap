// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ConnectModal > should render match DOM Snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="flex justify-center items-center overflow-x-hidden overflow-y-auto fixed inset-0 z-50 z-[51]"
    >
      <div
        class="flex w-full md:w-[640px] h-full md:h-[600px] border border-gray-700 md:rounded-sm bg-[#161B22]"
      >
        <div
          class="flex flex-col relative w-full"
        >
          <button
            class="absolute top-4 right-4 cursor-pointer"
            data-testid="cross"
          >
            <svg
              class="fill-muted hover:fill-blue-400 w-4 h-4"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.72 3.72a.75.75 0 011.06 0L8 6.94l3.22-3.22a.75.75 0 111.06 1.06L9.06 8l3.22 3.22a.75.75 0 11-1.06 1.06L8 9.06l-3.22 3.22a.75.75 0 01-1.06-1.06L6.94 8 3.72 4.78a.75.75 0 010-1.06z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
          <div
            class="py-2 px-4 border-b border-gray-700 text-2xl w-full"
          >
            Connect Wallet
          </div>
          <div
            class="w-full h-full md:h-[500px] overflow-x-hidden overflow-y-auto"
          >
            <div
              class="p-4"
            >
              <p
                class="text-xl pt-2"
              >
                Connect your wallet to continue with OpenQ. By connecting your wallet you agree with OpenQ's 
                <a
                  class="hover:underline text-blue-400"
                  href="/terms-of-use"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  terms of service
                </a>
                .
              </p>
              <button
                class="flex py-4 pl-4 my-8 w-full gap-8 btn-default"
              >
                <img
                  alt="metamask logo"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fwallet-logos%2Fmetamask.png&w=96&q=75"
                  srcset="/_next/image?url=%2Fwallet-logos%2Fmetamask.png&w=48&q=75 1x, /_next/image?url=%2Fwallet-logos%2Fmetamask.png&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
                <div
                  class="text-lg leading-loose"
                >
                  Metamask
                </div>
              </button>
              <button
                class="flex py-4 pl-4 mb-8 w-full gap-8 btn-default"
              >
                <img
                  alt="wallet connect logo"
                  class="rounded-full"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fwallet-logos%2Fwallet-connect.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fwallet-logos%2Fwallet-connect.jpg&w=48&q=75 1x, /_next/image?url=%2Fwallet-logos%2Fwallet-connect.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
                <div
                  class="leading-loose text-lg"
                >
                  WalletConnect
                </div>
              </button>
              <button
                class="flex py-4 pl-4 mb-8 w-full gap-8 btn-default"
              >
                <img
                  alt="wallet connect logo"
                  class="rounded-full"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fwallet-logos%2Fcoinbase-logo.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fwallet-logos%2Fcoinbase-logo.jpg&w=48&q=75 1x, /_next/image?url=%2Fwallet-logos%2Fcoinbase-logo.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
                <div
                  class="leading-loose text-lg"
                >
                  Coinbase
                </div>
              </button>
            </div>
          </div>
          <div
            class="flex justify-between items-center py-2 px-4 flex-wrap border-t border-gray-700 w-full "
          >
            <div
              class="my-1"
            />
            <div>
              <button
                class="btn-default w-full"
              >
                Close
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="bg-overlay z-10 fixed inset-0"
    />
  </div>
</DocumentFragment>
`;
